Failed code compilation including CLI11.hpp

**Describe the bug**
I have tried to compile the first CLI11 example  but It failed

**Environment**
- OS: Windows 64 bits
- Compiler: MSVC 143 VS 2022

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install CLI11 --triplet=x64-windows
2. See error: 
the error C1083 has been generated:
 It is not possible to include : 'impl/Encoding_inl.hpp': No such file or directory	
3. Example code
//+++++++++++++++++++++++++++++
#include "CLI/CLI.hpp"
#include <iostream>

int main(int argc, char** argv) {
    CLI::App app{ "App description" };

    // Define options
    int p = 0;
    app.add_option("-p", p, "Parameter");

    CLI11_PARSE(app, argc, argv);

    std::cout << "Parameter value: " << p << std::endl;
    return 0;
}
//++++++++++++++++++++++++++++
**Expected behavior**
The code must compile

--- a/ports/cli11/portfile.cmake
+++ b/ports/cli11/portfile.cmake
@@ -2,8 +2,10 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO CLIUtils/CLI11
     REF "v${VERSION}"
-    SHA512 965290d09977672d3bc3c57ca4b89a88c2c09461da6e866b18217d702d4d5a0977588fcb8fef1a3c3804e61ad80d276029f47469cc83dbfdc7021ee35f6b7269
+    SHA512 fdb61c430f5b99a9495fda7f94bfc8d0fb5360c99beeccbcb3b8918713579aac97fa0dcbce296065d9043f141a538c505919c9810fd1d192661e8b48b6a2637a
     HEAD_REF main
+    PATCHES
+        revert-1012-pkgconfig.diff
 )
 
 vcpkg_cmake_configure(

--- a/ports/cli11/portfile.cmake
+++ b/ports/cli11/portfile.cmake
@@ -21,4 +21,6 @@ vcpkg_cmake_config_fixup(CONFIG_PATH share/cmake/CLI11)
 vcpkg_fixup_pkgconfig()
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
 
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/CLI/CLI.hpp" "#pragma once" "#pragma once\n#ifndef CLI11_COMPILE\n#define CLI11_COMPILE\n#endif")
+
 vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/cli11/revert-1012-pkgconfig.diff
+++ b/ports/cli11/revert-1012-pkgconfig.diff
@@ -1,0 +1,10 @@
+diff --git a/cmake/CLI11GeneratePkgConfig.cmake b/cmake/CLI11GeneratePkgConfig.cmake
+index 7ed87f077..a9c5eb885 100644
+--- a/cmake/CLI11GeneratePkgConfig.cmake
++++ b/cmake/CLI11GeneratePkgConfig.cmake
+@@ -4,4 +4,4 @@ else()
+   configure_file("cmake/CLI11.pc.in" "CLI11.pc" @ONLY)
+ endif()
+ 
+-install(FILES "${PROJECT_BINARY_DIR}/CLI11.pc" DESTINATION "${CMAKE_INSTALL_DATADIR}/pkgconfig")
++install(FILES "${PROJECT_BINARY_DIR}/CLI11.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")

--- a/ports/cli11/vcpkg.json
+++ b/ports/cli11/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "cli11",
-  "version": "2.4.1",
+  "version": "2.4.2",
   "description": "CLI11 is a command line parser for C++11 and beyond that provides a rich feature set with a simple and intuitive interface.",
   "homepage": "https://github.com/CLIUtils/CLI11",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -1677,7 +1677,7 @@
       "port-version": 0
     },
     "cli11": {
-      "baseline": "2.4.1",
+      "baseline": "2.4.2",
       "port-version": 0
     },
     "clickhouse-cpp": {

--- a/versions/c-/cli11.json
+++ b/versions/c-/cli11.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "06ac9dc66d709da4eabe20f952b83b4bb7ae7dda",
+      "version": "2.4.2",
+      "port-version": 0
+    },
+    {
       "git-tree": "05a104e144c6cfa16b0a896502ef96f3ccbbdddc",
       "version": "2.4.1",
       "port-version": 0

